docs/proposal/20210823-gpu-support.md

```diff
@@ -0,0 +1,163 @@
+# GPU support in CAPV
+
+```text
+---
+title: GPU support in CAPV
+authors:
+  - "@geetikabatra"
+reviewers:
+  - "@vijaykumar"
+  - "@ankitaswamy"
+  - "@sonasingh46"
+  - "@pshail"
+  
+creation-date: 2021-08-23
+last-updated: 2021-08-25
+status: implementable
+```
+
+## Table of Contents
+
+* [GPU Support](#gpu-support)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1 - using vGPU](#story-1---using-vgpu)
+      * [Story 2 - GPU Direct implementation](#story-2---gpu-direct-implementation)
+      * [Story 3 - PCI passthrough for single node Customer](#story-3---pci-passthrough-for-single-node-customer)
+    * [Requirements](#requirements)
+      * [Functional Requirements](#functional-requirements)
+      * [Non-Functional Requirements](#non-functional-requirements)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Proposed Changes](#proposed-changes)
+      * [Controller Changes](#controller-changes)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+* CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+
+CAPV currently does not support GPU-based infrastructure.
+
+This proposal outlines ways to provision GPU support in CAPV. The proposed changes will maintain backward compatibility and maintain the existing behavior without any extra user configuration.
+
+## Motivation
+
+CAPV does not currently support GPU-accelerated workloads which have a limited addressable market. Competitively, all of the hyper-scale cloud providers (AWS, Azure, & GCP) offer GPU-accelerated virtualization and Kubernetes platforms. Many of them additionally offer robust vertically integrated AI/ML solutions building on open source technology. GPU support is a long-requested feature. With this capability, users can leverage GPU support.
+
+### Goals
+
+1. To enable GPU Support on Vsphere Kubernetes Cluster.
+2. To allow vGPU support for multi-node clusters.
+3. To allow GPU Direct support for multi-node clusters.
+4. To allow PCI passthrough support for single-node clusters.
```

Comment:
Does PCI passthrough necessarily limit things to a single node cluster? I also don't think goal 1 is needed as 2-4 covers that.

Reply:
Not sure about it. But there was a discussion going on that we need a Mellanox Network Operator for multi-node.